chore(gatsby-plugin-sitemap): Document `excludes` glob matching (#36690)
gatsbyjs/gatsby: packages/gatsby-plugin-sitemap/README.md

- `output` (string = `/sitemap`) Folder path where sitemaps are stored.
- `createLinkInHead` (boolean = true) Whether to populate the `<head>` of your site with a link to the sitemap.
- `entryLimit` (number = 45000) Number of entries per sitemap file. A sitemap index (as `sitemap-index.xml`) will always be created and multiple sitemaps are created for every `entryLimit` increment (e.g under 45000 entries only `sitemap-0.xml` will be created).

> - `excludes` (string[] = []) An array of paths to exclude from the sitemap. [Here are some examples](https://codepen.io/checkmark/pen/LYmOewV) of glob matching using [minimatch](https://github.com/isaacs/minimatch). While `excludes` is usually an array of strings it is possible to enter other data types into this array for custom filtering, but doing so will require customization of the [`filterPages`](#filterPages) function.

Reviewer: The Codepen can easily be changed/removed/deleted so it's best to only link to the official docs as they'll always be up to date

```suggestion
- `excludes` (string[] = []) An array of paths to exclude from the sitemap. You can use glob matching using [minimatch](https://github.com/isaacs/minimatch). While `excludes` is usually an array of strings it is possible to enter other data types into this array for custom filtering, but doing so will require customization of the [`filterPages`](#filterPages) function.
```

Author: Thank you for the review, and that makes sense. The suggested changes are good enough since they let the end user know that glob matching with minimatch is being used, and if they need to learn what that means it's at least a short search engine query away.
